ludwig/utils/trainer_utils.py

```diff
@@ -244,6 +255,40 @@ def log_metrics(self):
         return log_metrics
 
 
+@DeveloperAPI
+class WalltimeEarlyStopCallback(Callback):
+    """Callback that stops training when a timeout is reached and no improvement has been made."""
+
+    def __init__(self, timeout_s: float, early_stopping_steps: int = -1) -> None:
+        """Initializes the callback.
+
+        Args:
+            timeout_s: The timeout in seconds. Can be a float to allow for sub-second timeouts.
+            early_stopping_steps: The number of steps to wait before stopping. If -1, the timeout is the only
+                condition for stopping.
+        """
+        super().__init__()
+        self.timeout_delta = timedelta(seconds=timeout_s)
+        self.early_stopping_steps = early_stopping_steps
+
+    def should_early_stop(self, trainer, progress_tracker, is_coordinator):
+        """
+        Returns:
+            True if the timeout has been reached and no improvement has been made for the given early stopping steps.
+        """
+        # Time since last improvement > timeout
+        timed_out = (
```

Comment:
@jppgks are you testing this comparison? I don't think you can compare the difference between two datetimes against a float:

```
a = datetime.now()
b = datetime.now()
b-a > 10
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '>' not supported between instances of 'datetime.timedelta' and 'int'
```

I think you need `total_seconds()`. Also, let's make sure we're testing this.

Reply:
Ah, nevermind, see you're using a `timedelta` here.